lstchain/scripts/lstchain_r0_to_r0g.py

```
@@ -0,0 +1,206 @@
+from traitlets.config import Config
+
+import numpy as np
+from contextlib import ExitStack
+
+parser = argparse.ArgumentParser(description="Gain selector program (R0 to "
+                                             "R0G)")
+parser.add_argument('-f', '--R0-file', dest='input_file',
+                    type=str, help='Input R0 file name (of stream 1)')
+
+parser.add_argument('-o', '--output-dir', dest='output_dir',
+                    type=str, default='./',
+                    help='Output directory')
+
+# Range of waveform to be checked (for gain selection & heuristic FF
+# identification)
+SAMPLE_START = 3
+SAMPLE_END = 39
+
+def main():
+    args = parser.parse_args()
+
+    # Level of high gain (HG) required for switching to low gain (LG)
+    THRESHOLD = 3900
+
+    input_file = args.input_file
+    output_dir = args.output_dir
+
+    # First identify properly interleaved pedestals (also in case there are
+    # ucts jumps) and FF events (heuristically):
+    event_id, event_type = get_event_types(input_file)
+
+    event_id = np.array(event_id)
+    event_type_val = np.array([x.value for x in event_type])
+
+    logging.info('Identified event types and number of events:')
+    for j in np.unique(event_type_val):
+        logging.info(f'{EventType(j)}:, {np.sum(event_type_val == j)}\n')
+
+    # Now loop over the files (4 streams) again to perform the actual gain
+    # selection:
+    k = input_file.find('/LST-1.')
+    input_stream_names = [input_file[:k+7]+str(id_stream+1)+input_file[k+8:]
+                          for id_stream in range(4)]
+    output_stream_names = [output_dir + name[k:] for name in input_stream_names]
+
+    input_streams = []
+    for name in input_stream_names:
+        input_streams.append(protozfits.File(name, pure_protobuf=True))
+
+    try:
+        num_pixels = input_streams[0].CameraConfiguration[0].num_pixels
+        num_samples = input_streams[0].CameraConfiguration[
+            0].num_samples_nominal
+    except:
```

Bare except (see above). Should at least be `except Exception` to not catch `KeyboardInterrupt` and `SystemExit`.